data_steward/cdr_cleaner/cleaning_rules/base_cleaning_rule.py

```diff
@@ -118,6 +119,26 @@ class BaseCleaningRule(AbstractBaseCleaningRule):
     cleaning_class_list = List[AbstractBaseCleaningRule]
     TABLE_COUNT_QUERY = ''' SELECT COALESCE(COUNT(*), 0) AS row_count FROM `{dataset}.{table}` '''
 
+    # A query for dropping all empty sandbox tables
+    DROP_EMPTY_SANDBOX_TABLES_QUERY = JINJA_ENV.from_string("""
+    DECLARE i INT64 DEFAULT 0;
+    DECLARE tables DEFAULT (
+      SELECT
+        ARRAY_AGG(FORMAT("`%s.%s.%s`", project_id, dataset_id, table_id))
+      FROM
+        `{{project}}.{{dataset}}.__TABLES__`
+      WHERE
+        row_count = 0 AND table_id IN ({{table_ids}}));
+
+    LOOP
+      SET i = i + 1;
+      IF i > ARRAY_LENGTH(tables) THEN 
+        LEAVE; 
+      END IF;
+      EXECUTE IMMEDIATE '''DROP TABLE ''' || tables[ORDINAL(i)];
+    END LOOP
+    """)
+
```

Comment:
This function is only being used in `replace_standard_id_in_domain_tables` module and the `motor_vehicle_accident_concept_suppression` module. I think what you suggested would be a good alternative, we could potentially make this into a function instead of a class method for easy importing.

Reply:
You're right.  It might be a nice cleanup function.  I thought we had even considered implementing it as a separate cleaning rule, but it's not something I am highly prioritizing.  The sandboxes are primarily for us, to help with cleaning/de-identifying and manual debugging.  

Honestly, I'm not too concerned about the empty sandbox tables right now because that just tells us the rule executed and didn't find any data to sandbox.  It's nice to clean up, but you also lose a little information with it.

For now, I'd be satisfied just adding these as functions to the module so they are easily available.

Reply:
Okay, I can refactor this into a function. 

I do like what you said about keeping empty tables because the fact they are empty could be useful information as well. This is not super urgent now, just something for the future, I personally think it would be good for cleaning rules to have consistent behavior, either all the rules drop empty sandbox tables or keep all of the sandbox tables.